mm/2s2h/Rando/MiscBehavior/ClockShuffle.cpp

```diff
@@ -0,0 +1,765 @@
+#include "ClockShuffle.h"
+#include "Rando/Rando.h"
+#include "Rando/Logic/Logic.h"
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/CustomMessage/CustomMessage.h"
+#include <libultraship/bridge/consolevariablebridge.h>
+#include "2s2h/ShipUtils.h"
+
+extern "C" {
+#include "z64game.h"
+#include "overlays/gamestates/ovl_daytelop/z_daytelop.h"
+#include "overlays/actors/ovl_En_Test4/z_en_test4.h"
+#include "functions.h"
+}
+
+using namespace Rando::Logic;
+
+namespace Rando {
+
+// ============================================================================
+// CLOCK ITEM MANAGEMENT
+// ============================================================================
+
+namespace ClockItems {
+
+// Internal half-day indices
+enum ClockHalfIndex : int {
+    INVALID = -1,        // Invalid/not found/uninitialized half-day index
+    HALF_DAY1_DAY = 0,   // Day 1, 6:00 AM - 5:59 PM
+    HALF_DAY1_NIGHT = 1, // Day 1, 6:00 PM - 5:59 AM
+    HALF_DAY2_DAY = 2,   // Day 2, 6:00 AM - 5:59 PM
+    HALF_DAY2_NIGHT = 3, // Day 2, 6:00 PM - 5:59 AM
+    HALF_DAY3_DAY = 4,   // Day 3, 6:00 AM - 5:59 PM
+    HALF_DAY3_NIGHT = 5, // Day 3, 6:00 PM - 5:59 AM
+    TERMINAL_STATE = 6,  // Terminal state (fallback for invalid/end states)
+    HALF_COUNT = 6,      // Total number of regular half-days (0-5)
+};
+
+// Convert a rando item ID to its corresponding half-day index
+int GetHalfDayIndexFromClockItem(RandoItemId clockItemId) {
+    switch (clockItemId) {
+        case RI_CLOCK_DAY_1:
+            return HALF_DAY1_DAY;
+        case RI_CLOCK_NIGHT_1:
+            return HALF_DAY1_NIGHT;
+        case RI_CLOCK_DAY_2:
+            return HALF_DAY2_DAY;
+        case RI_CLOCK_NIGHT_2:
+            return HALF_DAY2_NIGHT;
+        case RI_CLOCK_DAY_3:
+            return HALF_DAY3_DAY;
+        case RI_CLOCK_NIGHT_3:
+            return HALF_DAY3_NIGHT;
+        default:
+            return INVALID; // Invalid item
```

Comment:
You could probably replace this whole thing with some enum arithmetic. Something like:
```suggestion
    return (clockItemId-RI_CLOCK_DAY_1);
```
Which itself might be cleaner to use as a macro rather than a function.

Reply:
I agree with using the arithmetic approach, but my preference would be to leave it as a function. Easier to debug.